backend/core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java

```diff
@@ -256,18 +264,26 @@ public void onBindingChanged(BindingConfigChangedEvent event) {
     if (!fullSynchronizationEnabled) {
       return;
     }
-    var configScopeId = event.getConfigScopeId();
+    var configScopeId = event.configScopeId();
     scopeSynchronizationTimestampRepository.clearLastSynchronizationTimestamp(configScopeId);
-    if (event.getPreviousConfig().isBound()) {
+    if (event.previousConfig().isBound()) {
       // when unbinding, we want to let future rebinds trigger a sync
-      var previousBinding = new Binding(requireNonNull(event.getPreviousConfig().getConnectionId()), requireNonNull(event.getPreviousConfig().getSonarProjectKey()));
+      var previousBinding = new Binding(requireNonNull(event.previousConfig().getConnectionId()), requireNonNull(event.previousConfig().getSonarProjectKey()));
       bindingSynchronizationTimestampRepository.clearLastSynchronizationTimestamp(previousBinding);
       branchSynchronizationTimestampRepository.clearLastSynchronizationTimestampIf(branchBinding -> branchBinding.getBinding().equals(previousBinding));
     }
-    var newConnectionId = event.getNewConfig().getConnectionId();
+    var newConnectionId = event.newConfig().getConnectionId();
     if (newConnectionId != null) {
-      synchronizeConnectionAndProjectsIfNeededAsync(newConnectionId,
-        List.of(new BoundScope(configScopeId, newConnectionId, requireNonNull(event.getNewConfig().getSonarProjectKey()))));
+      var projectKey = requireNonNull(event.newConfig().getSonarProjectKey());
+      var connection = connectionConfigurationRepository.getConnectionById(newConnectionId);
+      if (connection != null) {
+        if (connection.getKind() == ConnectionKind.SONARCLOUD) {
+          telemetryService.addBoundSQCProjectKey(projectKey);
+        } else {
+          telemetryService.addBoundSQSProjectKey(projectKey);
+        }
+      }
```

Comment:
Why doing this during the sync? Couldn't you listen to the `BindingConfigChangedEvent` from the telemetry service?